graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -318,16 +361,65 @@ private void addField(final String key, final Object value, final boolean isRequ
         }
 
         if (FIELD_TIMESTAMP.equals(trimmedKey) && value != null && value instanceof Date) {
-            fields.put(FIELD_TIMESTAMP, new DateTime(value));
+            final DateTime timestamp = new DateTime(value);
+            final Object previousValue = fields.put(FIELD_TIMESTAMP, timestamp);
+            updateSize(trimmedKey, timestamp, previousValue);
         } else if (value instanceof String) {
             final String str = ((String) value).trim();
 
             if (isRequiredField || !str.isEmpty()) {
-                fields.put(trimmedKey, str);
+                final Object previousValue = fields.put(trimmedKey, str);
+                updateSize(trimmedKey, str, previousValue);
             }
         } else if (value != null) {
-            fields.put(trimmedKey, value);
+            final Object previousValue = fields.put(trimmedKey, value);
+            updateSize(trimmedKey, value, previousValue);
+        }
+    }
+
+    private void updateSize(String fieldName, Object newValue, Object previousValue) {
+        // don't count internal fields
+        if (GRAYLOG_FIELDS.contains(fieldName)) {
+            return;
+        }
+        long newValueSize = 0;
+        long oldValueSize = 0;
+        final long oldSize = sizeCounter.getCount();
```

Comment:
The `oldSize` variable is only used in the TRACE log message and could be moved into the `isTraceEnabled` block to avoid the variable allocation on every call.

Reply:
But then we need to delay updating the `sizeCounter` and I'm not sure it's really worth the effort for a stack variable.